src/mlpack/tests/ann/layer/repeat.cpp

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file tests/ann/layer/repeat.cpp
+ * @author Adam Kropp
+ *
+ * Tests the repeat layer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann.hpp>
+
+#include "../../test_catch_tools.hpp"
+#include "../../catch.hpp"
+#include "../../serialization.hpp"
+#include "../ann_test_tools.hpp"
+
+using namespace mlpack;
+
+/**
+ * Simple test for Repeat layer along axis 0.
+ */
+TEST_CASE("RepeatTestCase0", "[ANNLayerTest]")
+{
+  // Input will be 4 x 3.
+  arma::mat input = arma::mat(4, 3, arma::fill::randn);
```

Comment:
Can we have an additional float 32 test in here, it should be very simple like this :

```c++
// Test all constructor variants.  This test is more about checking that all
// variants compile correctly than anything else.
TEMPLATE_TEST_CASE("LogisticRegressionAllConstructorsTest",
    "[LogisticRegressionTest]", arma::mat, arma::fmat)
```
Then you define MatType inside instead of `arma::mat`, the full example can be found here: 

https://github.com/mlpack/mlpack/pull/3564/files/665d3457cc00fc1efb92e8c51c1f43175b183b90#diff-dc91ef45b9c554294e4b7b68c50303d101fa2f5f9bc2867cf63f6019e81b5d67